src/codecs/avif/encoder.rs

```diff
@@ -124,7 +119,7 @@ impl<W: Write> AvifEncoder<W> {
     }
 
     fn encode_as_img<'buf>(
-        &'buf mut self,
+        fallback: &'buf mut Vec<u8>,
```

Comment:
At this point, `fallback` could just be allocated locally in this function.

Reply:
I thought so initially, but it turns out that doesn't work. The return value is a reference. When no colour conversion is required that reference is into `data`, which is borrowed anyway, so that's fine. When the input isn't RGB(A), though, the reference is into `fallback`, which means it can't be deallocated on function return.

There's _probably_ a way around this, but `ravif` is very specific about the type of its input data and getting that to align with maybe-owned-maybe-borrowed values is beyond my Rust experience to solve rapidly. :)